src/test/java/test/annotationtransformer/AnnotationTransformerTest.java

```diff
@@ -140,8 +141,20 @@ private void runTest(IAnnotationTransformer transformer,
 
     Assert.assertEquals(results.size(), 1);
     Assert.assertEquals(name, results.get(0).getMethod().getMethodName());
+    Assert.assertTrue(MySuiteListener.triggered);
   }
 
+  @Test
+  public void verifyListenerAnnotationTransformerClass() {
+    MySuiteListener.triggered = false;
+    TestNG tng = new TestNG();
+    tng.setVerbose(0);
+    tng.setAnnotationTransformer(new MyListenerTransformer());
+    tng.setTestClasses(new Class[]{AnnotationTransformerClassSampleTest.class});
+
+    tng.run();
+    Assert.assertFalse(MySuiteListener.triggered);
```

Comment:
Maybe you could make the test stronger by actually having the transformer set the listeners to another class and assert that this class was called instead of just testing the original listener wasn't called?